src/main/java/org/damap/base/rest/invenio_damap/InvenioDAMAPResource.java

```diff
@@ -0,0 +1,171 @@
+import org.damap.base.repo.AccessRepo;
+import org.damap.base.rest.dmp.domain.DmpDO;
+import org.damap.base.rest.dmp.domain.DmpListItemDO;
+import org.damap.base.rest.dmp.service.DmpService;
+import org.damap.base.rest.madmp.dto.Dataset;
+import org.damap.base.security.SecurityService;
+import org.damap.base.validation.AccessValidator;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+@Path("/api/madmps")
+@RequestScoped
+@Produces(MediaType.APPLICATION_JSON)
+@JBossLog
+public class InvenioDAMAPResource {
+
+  @Inject
+  public InvenioDAMAPResource(
+      SecurityService securityService,
+      AccessValidator accessValidator,
+      AccessRepo accessRepo,
+      DmpService dmpService,
+      InvenioDAMAPService invenioDAMAPService) {
+    this.securityService = securityService;
+    this.accessValidator = accessValidator;
+    this.accessRepo = accessRepo;
+    this.dmpService = dmpService;
+    this.invenioDAMAPService = invenioDAMAPService;
+  }
+
+  SecurityService securityService;
+  AccessValidator accessValidator;
+  AccessRepo accessRepo;
+  DmpService dmpService;
+  InvenioDAMAPService invenioDAMAPService;
+
+  @ConfigProperty(name = "invenio.shared-secret")
+  String sharedSecret;
+
+  @Inject JWTParser parser;
+
+  /*
+   * Maybe make it configurable?
+   * Could be more elaborate but should suffice for PoC
+   * Could also go into the {@link org.damap.base.security.SecurityService},
+   * but keeping everything together for PoC
+   *
+   */
+  private JsonWebToken validateAuthHeader(HttpHeaders headers) {
+    String jwt_token = headers.getHeaderString("X-Auth");
+
+    if (jwt_token != null && !jwt_token.isEmpty()) {
+      try {
+        JsonWebToken jwt = parser.verify(jwt_token, sharedSecret);
+
+        long exp = jwt.getExpirationTime();
+        long currentTime = System.currentTimeMillis() / 1000;
+
+        if (currentTime >= exp) throw new UnauthorizedException("Token expired.");
+
+        return jwt;
+      } catch (ParseException e) {
+        return null;
+      }
+    }
+    return null; // No Authorization header or not in the correct format
+  }
+
+  // Checks provided jwt.
+  private JsonWebToken checkIfUserIsAuthorized(HttpHeaders headers) {
+
+    JsonWebToken jwt = validateAuthHeader(headers);
+    boolean authorized = jwt != null;
+
+    if (!authorized) {
+      throw new UnauthorizedException("User unauthorized.");
+    }
+    return jwt;
+  }
+
+  // Resolves user identity based on the DMPs and returns dmp list and list of
+  // identifiers if successful.
+  private SimpleEntry<List<DmpListItemDO>, List<String>> resolveDmpsAndIds(JsonObject identifiers) {
+    List<DmpListItemDO> personDmpList = null;
+    List<String> matchingIdentifiers = new ArrayList<>();
+
+    for (String key : identifiers.keySet()) {
+      String identifier = identifiers.getString(key);
+      List<DmpListItemDO> dmps = dmpService.getDmpListByPersonId(identifier);
```

Comment:
general note: We should __really__ try to get some sort of user management into DAMAP. Otherwise, we always have to fetch DMPs :)

Reply:
Agree with that, this was rather painful :smiley: This should come as a future work though :slightly_smiling_face: